src/sas/sascalc/invariant/invariant.py

```diff
@@ -921,45 +945,52 @@ def get_volume_fraction_with_error(self, contrast, extrapolation=None):
+            uncertainty = math.fabs((k * self._qstar_err)
+                                    / (self._qstar * math.sqrt(1 - 4 * k)))
 
         return volume, uncertainty
 
     def get_surface_with_error(self, contrast, porod_const, extrapolation=None):
         """
-        Compute uncertainty of the surface value as well as the surface value.
-        The uncertainty is given as follow: ::
+        As of SasView 4.3 and 5.0.2, the specific surface is computed directly
+        from the contrast and porod_constant wich are currently user inputs
+        with no option for any uncertainty so no uncertainty can be calculated.
+        However we include the uncertainty computation for future use if and
+        when these values get an uncertainty. This is given as:  ::
 
-            dS = porod_const *2*pi[( dV -2*V*dV)/q_star
-                 + dq_star(v-v**2)
+                ds/s = sqrt[(dcp/cp)**2 + 2* (dcontrast/contrast)**2]
 
-            q_star: the invariant value
-            dq_star: the invariant uncertainty
-            V: the volume fraction value
-            dV: the volume uncertainty
+            which gives (this should be checked before using in anger)
+                ds = sqrt((dporod_const)**2 + 2 * (porod_const *
+                          dcontrast / contrast)**2) / (2 * pi * contrast**2)
 
-        :param contrast: contrast value
-        :param porod_const: porod constant value
-        :param extrapolation: string to apply optional extrapolation
+        :param contrast: contrast value eventually with the error
+        :param porod_const: porod constant value eventually with the error
+        :param extrapolation: string to apply optional extrapolation. This will
+               only be needed if and when the contrast term is calculated from
+               the invariant.
 
-        :return S, dS: the surface, with its uncertainty
+        :return s, ds: the surface, with its uncertainty
         """
-        # We get the volume fraction, with error
-        #   get_volume_fraction_with_error calls get_volume_fraction
-        #   get_volume_fraction calls get_qstar
-        #   which computes Qstar and dQstar
-        v, dv = self.get_volume_fraction_with_error(contrast, extrapolation)
 
-        s = self.get_surface(contrast=contrast, porod_const=porod_const,
-                             extrapolation=extrapolation)
-        ds = porod_const * 2 * math.pi * ((dv - 2 * v * dv) / self._qstar\
-                 + self._qstar_err * (v - v ** 2))
+        # until contrast and porod_constant are given with uncertainties set
+        # them to 0
+        dcontrast = 0
+        # If and when dporod_const exists remember it likely needs unit
+        # conversion from cm^-1 A^-4 to A^-5.
+        dporod_const = 0 * 1e-8
+        _porod_const = porod_const *1e-8
+
+        s = self.get_surface(contrast=contrast, porod_const=porod_const)
+        ds = math.sqrt((dporod_const)**2 + 2 * (_porod_const\
```

Comment:
Extra parens around `dporod_const` don't do anything.  And again, the `\` isn't needed here,

Reply:
fixing point one ... and point 2 but not sure that helps without violating lint rules? I either need a
third line which will require a \ apparently, or I need to move the start of the second line much farther forward that it should be given the parenthesis?  suggestions?

Reply:
can this be set as resolved?